src/boardState.ts

```diff
@@ -0,0 +1,239 @@
+
+            while(!firstUnresolved){
+                if(shotImpliedPlacementsList[index].resolved == false){
+                    firstUnresolved = shotImpliedPlacementsList[index];
+                }
+                index++;
+            }
+            
+            let nextShip:ShipPlacement
+            if(firstUnresolved.allValidPlacements.length>0){
+                if(!exhaustive){
+                    nextShip = firstUnresolved.pickRandomPlacement();
+                }
+                else{
+                    let nextStep: number = 0;
+                    if(pathToFollow.length > currentPathIndex){ //if the length is good
+                        nextStep = pathToFollow[currentPathIndex];
+                        if(nextStep > firstUnresolved.allValidPlacements.length-1){
+                            let nextPath:number[] = currentPath;
+                            nextPath[nextPath.length-1]++;
+                            return {state: new BoardState([]),found:false, nextPath: nextPath};
+                        }
+                    }
+                    currentPathIndex++;
+                    currentPath.push(nextStep);
+                    nextShip = firstUnresolved.allValidPlacements[nextStep]
+                }
+            }
+            else{
+                if(!exhaustive)
+                {
+                    return {state: new BoardState([]), found:false, nextPath: []};
+                }else{
+                    let nextPath:number[] = currentPath;
+                    nextPath[nextPath.length-1]++;
+                    return {state: new BoardState([]),found:false, nextPath: nextPath};
+                }
+            }
+
+            let occupiedPositions: Array<Position> = nextShip.getOccupiedPositions();
+
+            possibleShips.push(nextShip);
+            numberRemaining[nextShip.length]--;
+            for(var i=0;i<validShipPositions.length; i++){
+                validShipPositions[i].removePossibilities(nextShip);
+            }
+            
+            for(var j=0; j<shotImpliedPlacementsList.length;j++){
+                for(var i=0; i<occupiedPositions.length;i++){
+                    
+                    if((!shotImpliedPlacementsList[j].resolved)&& Position.isEqual(shotImpliedPlacementsList[j].shot.Position, occupiedPositions[i])){
+                        
+                        numberUnresolved--;
+                        shotImpliedPlacementsList[j].resolved = true;
+                    }
+                    
+                }
+                shotImpliedPlacementsList[j].updateRemainingShips(numberRemaining);
+                shotImpliedPlacementsList[j].removeInvalidPositions(validShipPositions);
+            }
+
+            shotImpliedPlacementsList.sort((a,b,)=>{return a.allValidPlacements.length - b.allValidPlacements.length});
+        }
+
+
+        //for whatever ships have yet to be placed (needs to be checked)
+        //place those ships randomly
```

Comment:
Again, if you're using comments to explain what's going on, it's a really strong reason to go back and try and extract things into named classes/methods.